catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveCatalogOperations.java

```diff
@@ -135,11 +153,80 @@ public void initialize(Map<String, String> conf) throws RuntimeException {
     // and gravitinoConfig will be passed to Hive config, and gravitinoConfig has higher priority
     mergeConfig.forEach(hadoopConf::set);
     hiveConf = new HiveConf(hadoopConf, HiveCatalogOperations.class);
+    UserGroupInformation.setConfiguration(hadoopConf);
+
+    initKerberosIfNecessary(conf, hadoopConf);
 
     this.clientPool =
         new CachedClientPool(getClientPoolSize(conf), hiveConf, getCacheEvictionInterval(conf));
   }
 
+  private void initKerberosIfNecessary(Map<String, String> conf, Configuration hadoopConf) {
+    if (UserGroupInformation.AuthenticationMethod.KERBEROS
+        == SecurityUtil.getAuthenticationMethod(hadoopConf)) {
+      try {
+        File file = new File("tmp");
+        if (!file.exists()) {
+          file.mkdir();
+        }
+
+        File keyTabFile = new File(String.format("tmp/gravitino-%s-keytab", entity.id()));
+        keyTabFile.deleteOnExit();
+        if (keyTabFile.exists() && !keyTabFile.delete()) {
+          LOG.error("Fail to delete key tab file {}", keyTabFile.getAbsolutePath());
```

Comment:
Should you throw an exception here, or continue to execute your code?

Reply:
Fixed.